File: problem_sets/sentimental/mario-less/mario.py
from cs50 import get_int

# Prompt user for height until a valid input is received
while True:
    height = get_int("Height: ")
    if 1 <= height <= 8:
        break

# Print the half-pyramid
for i in range(1, height + 1):
    print(" " * (height - i) + "#" * i)

File: problem_sets/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;

    // Prompt user for height between 1 and 8 (inclusive)
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Loop through each row
    for (int row = 1; row <= height; row++)
    {
        // Print spaces for left pyramid
        for (int space = 0; space < height - row; space++)
        {
            printf(" ");
        }

        // Print hashes for left pyramid
        for (int hash = 0; hash < row; hash++)
        {
            printf("#");
        }

        // Print gap
        printf("  ");

        // Print hashes for right pyramid
        for (int hash = 0; hash < row; hash++)
        {
            printf("#");
        }

        // Move to the next line
        printf("\n");
    }
}
